include/openmc/cell.h

```diff
@@ -249,6 +249,32 @@ class Cell {
   std::unordered_map<int32_t, vector<int32_t>> get_contained_cells(
     int32_t instance = 0, Position* hint = nullptr) const;
 
+  //! Determine the material index corresponding to a specific cell instance,
+  //! taking into account presence of distribcell material
+  //! \param[in] instance of the cell
+  //! \return material index
+  int32_t material(int32_t instance) const
+  {
+    if (material_.size() > 1) {
```

Comment:
Could you add a comment explaining here and in the below method that this branch is here because we allow either one material for each distribcell, and if that's the case, it's assumed to apply to every cell?